In [23]:
"""O objetivo deste trabalho é verificar se é possível definir um perfil para as vítimas.
O objetivo central verificar se nos ataques de tubarão há alguma diferença entre o perfil dos sexo das vítimas, 
assim como há em acidentes de trânsito(https://www.cbsnews.com/news/men-vs-women-who-are-safer-drivers/) 
e em crimes violentos (https://www.thejournal.ie/readme/violence-against-women-1959171-Feb2015/)"""

'O objetivo deste trabalho é verificar se é possível definir um perfil para as vítimas.\nO objetivo central verificar se nos ataques de tubarão há alguma diferença entre o perfil dos sexo das vítimas, \nassim como há em acidentes de trânsito(https://www.cbsnews.com/news/men-vs-women-who-are-safer-drivers/) \ne em crimes violentos (https://www.thejournal.ie/readme/violence-against-women-1959171-Feb2015/)'

In [24]:
import pandas as pd
import numpy as np

In [25]:
df=pd.read_csv('attacks.csv', encoding="latin-1")

In [26]:
df.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   ...  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
2                    Habush, Oahu      Surfing      John Denges    M  ...   
3              Arrawarra Headland      Surfing             male    M  ...   
4                        La Ticla  Free diving   Gustavo Ramos     M  ...   

          Species           Investigator or Source                       pdf  \
0      White shark                R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1              NaN  K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2              NaN  K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3        2 m shark                  B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4  Tiger shark, 3m                       A .Kipper      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2 original order Unnamed: 22 Unnamed: 23  
0    2018.06.25         6303.0         NaN         NaN  
1    2018.06.18         6302.0         NaN         NaN  
2    2018.06.09         6301.0         NaN         NaN  
3    2018.06.08         6300.0         NaN         NaN  
4    2018.06.04         6299.0         NaN         NaN  

[5 rows x 24 columns]

In [27]:
df.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   ...  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
2                    Habush, Oahu      Surfing      John Denges    M  ...   
3              Arrawarra Headland      Surfing             male    M  ...   
4                        La Ticla  Free diving   Gustavo Ramos     M  ...   

          Species           Investigator or Source                       pdf  \
0      White shark                R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1              NaN  K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2              NaN  K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3        2 m shark                  B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4  Tiger shark, 3m                       A .Kipper      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2 original order Unnamed: 22 Unnamed: 23  
0    2018.06.25         6303.0         NaN         NaN  
1    2018.06.18         6302.0         NaN         NaN  
2    2018.06.09         6301.0         NaN         NaN  
3    2018.06.08         6300.0         NaN         NaN  
4    2018.06.04         6299.0         NaN         NaN  

[5 rows x 24 columns]

In [28]:
df.shape

(25723, 24)

In [29]:
#Retirando todas as linhas que não tem informação alguma
df.dropna(how="all" , inplace=True)

In [30]:
df.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   ...  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
2                    Habush, Oahu      Surfing      John Denges    M  ...   
3              Arrawarra Headland      Surfing             male    M  ...   
4                        La Ticla  Free diving   Gustavo Ramos     M  ...   

          Species           Investigator or Source                       pdf  \
0      White shark                R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1              NaN  K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2              NaN  K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3        2 m shark                  B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4  Tiger shark, 3m                       A .Kipper      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2 original order Unnamed: 22 Unnamed: 23  
0    2018.06.25         6303.0         NaN         NaN  
1    2018.06.18         6302.0         NaN         NaN  
2    2018.06.09         6301.0         NaN         NaN  
3    2018.06.08         6300.0         NaN         NaN  
4    2018.06.04         6299.0         NaN         NaN  

[5 rows x 24 columns]

In [31]:
df.shape

(8703, 24)

In [32]:
# Verificando o novo data frame
df.describe

<bound method NDFrame.describe of       Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
8698            0          NaN     NaN         NaN        NaN   
8699            0          NaN     NaN         NaN        NaN   
8700            0          NaN     NaN         NaN        NaN   
8701            0          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, G

In [33]:
df.head()

Case Number         Date    Year        Type    Country             Area  \
0  2018.06.25  25-Jun-2018  2018.0     Boating        USA       California   
1  2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA          Georgia   
2  2018.06.09  09-Jun-2018  2018.0     Invalid        USA           Hawaii   
3  2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA  New South Wales   
4  2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO           Colima   

                         Location     Activity             Name Sex   ...  \
0     Oceanside, San Diego County     Paddling      Julie Wolfe    F  ...   
1  St. Simon Island, Glynn County     Standing  Adyson McNeely     F  ...   
2                    Habush, Oahu      Surfing      John Denges    M  ...   
3              Arrawarra Headland      Surfing             male    M  ...   
4                        La Ticla  Free diving   Gustavo Ramos     M  ...   

          Species           Investigator or Source                       pdf  \
0      White shark                R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1              NaN  K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2              NaN  K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3        2 m shark                  B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4  Tiger shark, 3m                       A .Kipper      2018.06.04-Ramos.pdf   

                                        href formula  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...   
1  http://sharkattackfile.net/spreadsheets/pdf_di...   
2  http://sharkattackfile.net/spreadsheets/pdf_di...   
3  http://sharkattackfile.net/spreadsheets/pdf_di...   
4  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                href Case Number.1  \
0  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.18   
2  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.09   
3  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.08   
4  http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.04   

  Case Number.2 original order Unnamed: 22 Unnamed: 23  
0    2018.06.25         6303.0         NaN         NaN  
1    2018.06.18         6302.0         NaN         NaN  
2    2018.06.09         6301.0         NaN         NaN  
3    2018.06.08         6300.0         NaN         NaN  
4    2018.06.04         6299.0         NaN         NaN  

[5 rows x 24 columns]

In [34]:
list(df)

['Case Number',
 'Date',
 'Year',
 'Type',
 'Country',
 'Area',
 'Location',
 'Activity',
 'Name',
 'Sex ',
 'Age',
 'Injury',
 'Fatal (Y/N)',
 'Time',
 'Species ',
 'Investigator or Source',
 'pdf',
 'href formula',
 'href',
 'Case Number.1',
 'Case Number.2',
 'original order',
 'Unnamed: 22',
 'Unnamed: 23']

In [35]:
# Retirando todas as colunas que não serão importantes na análise
df.drop(["Case Number","Date","Year","Area","Location","Activity","Name","Time","Species ",'Investigator or Source','pdf','href formula','Case Number.1','Case Number.2','original order','Unnamed: 22','Unnamed: 23',"href"], inplace=True , axis=1)

In [36]:
#Verificação do novo dataframe
df.head()

Type    Country Sex   Age  \
0     Boating        USA    F   57   
1  Unprovoked        USA    F   11   
2     Invalid        USA    M   48   
3  Unprovoked  AUSTRALIA    M  NaN   
4    Provoked     MEXICO    M  NaN   

                                              Injury Fatal (Y/N)  
0  No injury to occupant, outrigger canoe and pad...           N  
1                         Minor injury to left thigh           N  
2       Injury to left lower leg from surfboard skeg           N  
3                          Minor injury to lower leg           N  
4  Lacerations to leg & hand shark PROVOKED INCIDENT           N

In [37]:
# Contagem de acidentes por sexo
df['Sex '].value_counts()

M      5094
F       637
M         2
N         2
.         1
lli       1
Name: Sex , dtype: int64

Ainda que todos os valores inválidos para o sexo fossem considerados como femininos, 80% das vítimas dos ataques são homens. Considerando só os casos em que há documentação, 88% são homens.

In [38]:
# Proporção dos acidentes por sexo
df['Sex '].value_counts(normalize=True)

M      0.887921
F      0.111034
M      0.000349
N      0.000349
.      0.000174
lli    0.000174
Name: Sex , dtype: float64

In [39]:
#Contagem dos acidentes fatais
df["Fatal (Y/N)"].value_counts()

N          4293
Y          1388
UNKNOWN      71
 N            7
N             1
2017          1
M             1
y             1
Name: Fatal (Y/N), dtype: int64

In [40]:
# Criação de um novo dataframe(pof) com somente os acidentes que são fatais
pof=df[df['Fatal (Y/N)'] == "Y"]

In [41]:
#Contagem dos acidentes fatais por sexo
pof["Sex "].value_counts()

M    1172
F     107
.       1
N       1
Name: Sex , dtype: int64

In [42]:
# Acidentes fatais, por sexo, em proporção
pof["Sex "].value_counts(normalize=True)

M    0.914910
F    0.083528
.    0.000781
N    0.000781
Name: Sex , dtype: float64

No caso das vítimas fatais, a proporção masculina é ainda maior, variando entre 84% e 91%, dependendo das simplificações.